src/js/components/channels/twilio-channel-content.jsx

```diff
@@ -1,34 +1,134 @@
 import React, { Component } from 'react';
+import { updateTwilioAttributes, resetTwilioAttributes } from '../../services/integrations-service';
 
 import { ReactTelephoneInput } from '../../lib/react-telephone-input';
 
-export class TwilioChannelContent extends Component {
-    state = {
-        numberValid: false
+import { connect } from 'react-redux';
+import isMobile from 'ismobilejs';
+
+export class TwilioChannelContentComponent extends Component {
+
+    linkTwilioNumber = () => {
+        const {state: {formattedNumber}} = this.refs.telInput;
+        updateTwilioAttributes({
+            number: formattedNumber,
+            linkState: 'pending'
+        });
     }
 
+    onRetry = () => {
+        updateTwilioAttributes({
+            linkState: 'linked'
+        });
+    }
+
+    onChangeNumber = () => {
+        updateTwilioAttributes({
+            linkState: 'unlinked',
+            number: '',
+            numberValid: false
+        });
+    }
+
+    onStartTexting = () => {
+        updateTwilioAttributes({
+            linkState: 'linked'
+        });
+    }
+
+    onSendText = () => {
+        updateTwilioAttributes({
+            linkState: 'linked'
+        });
+    }
 
     onNumberValid = () => {
-        this.setState({
+        updateTwilioAttributes({
             numberValid: true
         });
     }
 
     onNumberInvalid = () => {
-        this.setState({
+        updateTwilioAttributes({
             numberValid: false
         });
     }
 
+    componentWillUnmount() {
+        resetTwilioAttributes();
+    }
+
     render() {
-        return <div>
-                   <ReactTelephoneInput ref='telInput'
-                                        defaultCountry='ca'
-                                        onChange={ this.handleInputChange }
-                                        onValid={ this.onNumberValid }
-                                        onInvalid={ this.onNumberInvalid }
-                                        preferredCountries={ ['ca', 'us'] }
-                                        onBlur={ this.handleInputBlur } />
-               </div>;
+        let iconStyle = {};
+        if (this.props.settings.linkColor) {
+            iconStyle = {
+                color: `#${this.props.settings.linkColor}`
+            };
+        }
+
+        const linkButton = this.props.numberValid ? <button className='btn btn-sk-primary'
+                                                            onClick={ this.linkTwilioNumber }>
+                                                        Continue
+                                                    </button> : '';
+        const unlinkedComponent = <div>
+                                      <ReactTelephoneInput ref='telInput'
```

Comment:
string refs are legacy since React 15 (https://facebook.github.io/react/docs/more-about-refs.html). The new way to use them is to use callbacks like this `<ReactTelephoneInput ref={(c) => this._telInput = c} /> ` and then use `this._telInput` directly.